app/controllers/organizations_controller.rb

```diff
@@ -30,4 +31,30 @@ def create
     @organization = e.record
     render :new
   end
+
+  def destroy
+    @organization = Organization.find params[:id]
+
+    if @organization.soft_delete
+      flash[:success] = "Organization '#{@organization.name}' deleted!"
+    else
+      flash[:error] = <<-eos
+        '#{@organization.name}' was unable to be deleted. We found the following open orders:
+        #{@organization.open_orders.map(&:id).to_sentence}
```

Comment:
:+1: w00t!

Reply:
Something to consider though, it could be possible for something else to fail and then this message would show up potentially without any errors. If you want to avoid that scenario, you could create an exception class to throw when there are open orders, and then use `save!` instead of `save`, then catch the specific exception to flash the error. Other errors you could flash a more generic error message, or even just let them cause a regular server error.

Reply:
Unless I'm wrong a error would raise and neither flash would appear.
```
2.2.4 :007 > def foo
2.2.4 :008?>     raise RuntimeError
2.2.4 :009?>   end
 => :foo
2.2.4 :010 >
2.2.4 :011 >   if foo
2.2.4 :012?>     puts true
2.2.4 :013?>   else
2.2.4 :014 >       puts false
2.2.4 :015?>   end
RuntimeError: RuntimeError
	from (irb):8:in `foo'
	from (irb):11
```

Reply:
this has been done 👍